src/lib/autocontainers.ts

```diff
@@ -158,12 +173,86 @@ export class AutoContain implements IAutoContain {
         }
     }
 
-    // Parses autocontain directives and returns valid cookieStoreIds or errors.
-    parseAucons = async (details): Promise<string> => {
+    // Checks to see if there are any other container-related extensions and avoids getting into
+    // fights with them.
+    checkOtherExtensionsHavePriority = async (
+        details: browser.webRequest.IDetails,
+    ): Promise<boolean> => {
+        // The checks for each extension can be done in parallel.
+        let priorities = await Promise.all([
+            this.checkMACPriority(details),
+            this.checkTempContainersPriority(details),
+        ])
+        return priorities.some(t => t)
+    }
+
+    checkMACPriority = async (
+        details: browser.webRequest.IDetails,
+    ): Promise<boolean> => {
+        if (
+            !ExtensionInfo.getExtensionEnabled(
+                ExtensionInfo.KNOWN_EXTENSIONS.multi_account_containers,
+            )
+        ) {
+            // It can't take priority if it's not enabled.
+            return false
+        }
+
+        // Do not handle urls that are claimed by the multi-account
+        // containers extension. Code from
+        // https://github.com/mozilla/multi-account-containers/wiki/API
+        let macAssignment = await browser.runtime
+            .sendMessage(
+                ExtensionInfo.KNOWN_EXTENSIONS.multi_account_containers,
+                {
+                    method: "getAssignment",
+                    url: details.url,
+                },
+            )
+            .catch(error => {
+                // TODO: figure out a good way to recover?
```

Comment:
I don't know if there's any state to recover if MAC fails. Shouldn't we just assume that if it does, aucons take over?

We should also add logging here I feel.

Reply:
Added some logging.

I was actually debating whether continuing was the right thing to do here. If we've gotten to this point, then theoretically MAC is installed; if this error is a transient communications failure or some bug in the communication, then continuing could lead to MAC and aucons fighting over their containers and looping infinitely.